core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -71,6 +71,25 @@ public boolean performWriteSkewCheck(DataContainer container, PersistenceManager
       return InequalVersionComparisonResult.EQUAL == result;
    }
 
+   private EntryVersion getCurrentEntryVersion(DataContainer container, PersistenceManager persistenceManager, TxInvocationContext ctx, VersionGenerator versionGenerator, TimeService timeService) {
+      EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
+      // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
```

Comment:
The second command says that access to persistence belongs to interceptors, as these modify the state of context, commands and other components. Methods withing `CacheEntry` should access only the entry itself.